panther_hardware_interfaces/include/panther_hardware_interfaces/roboteq_driver.hpp

```diff
@@ -0,0 +1,235 @@
+{
+
+struct RoboteqMotorState
+{
+  std::int32_t pos;
+  std::int32_t vel;
+  std::int32_t current;
+};
+
+struct RoboteqDriverFeedback
+{
+  RoboteqMotorState motor_1;
+  RoboteqMotorState motor_2;
+
+  std::uint8_t fault_flags;
+  std::uint8_t script_flags;
+  std::uint8_t runtime_stat_flag_motor_1;
+  std::uint8_t runtime_stat_flag_motor_2;
+
+  timespec timestamp;
+};
+
+// todo: heartbeat timeout (on hold - waiting for decision on changing to PDO)
+/**
+ * @brief Implementation of FiberDriver for Roboteq drivers
+ */
+class RoboteqDriver : public lely::canopen::FiberDriver
+{
+public:
+  using FiberDriver::FiberDriver;
+
+  RoboteqDriver(
+    const std::shared_ptr<lely::ev::Executor> & exec,
+    const std::shared_ptr<lely::canopen::AsyncMaster> & master, const std::uint8_t id,
+    const std::chrono::milliseconds & sdo_operation_timeout_ms);
+
+  /**
+   * @brief Trigger boot operations
+   */
+  bool Boot();
+
+  /**
+   * @brief Waits until the booting procedure finishes
+   *
+   * @exception std::runtime_error if boot fails
+   */
+  bool WaitForBoot();
+
+  bool IsBooted() const { return booted_.load(); }
+
+  bool IsCANError() const { return can_error_.load(); }
+
+  /**
+   * @exception std::runtime_error if operation fails
+   */
+  std::int16_t ReadTemperature();
+
+  /**
+   * @exception std::runtime_error if operation fails
+   */
+  std::uint16_t ReadVoltage();
+
+  /**
+   * @exception std::runtime_error if operation fails
+   */
+  std::int16_t ReadBatAmps1();
+
+  /**
+   * @exception std::runtime_error if operation fails
+   */
+  std::int16_t ReadBatAmps2();
+
+  /**
+   * @brief Reads all the PDO data returned from Roboteq (motors feedback, error flags) and saves
+   * current timestamp
+   */
+  RoboteqDriverFeedback ReadRoboteqDriverFeedback();
+
+  /**
+   * @param cmd command value in the range [-1000, 1000]
+   * @exception std::runtime_error if operation fails
+   */
+  void SendRoboteqCmdChannel1(const std::int32_t cmd);
+
+  /**
+   * @param cmd command value in the range [-1000, 1000]
+   * @exception std::runtime_error if operation fails
+   */
+  void SendRoboteqCmdChannel2(const std::int32_t cmd);
+
+  /**
+   * @exception std::runtime_error if any operation returns error
+   */
+  void ResetRoboteqScript();
+
+  /**
+   * @exception std::runtime_error if any operation returns error
+   */
+  void TurnOnEstop();
+
+  /**
+   * @exception std::runtime_error if any operation returns error
+   */
+  void TurnOffEstop();
+
+  /**
+   * @exception std::runtime_error if any operation returns error
+   */
+  void TurnOnSafetyStopChannel1();
+
+  /**
+   * @exception std::runtime_error if any operation returns error
+   */
+  void TurnOnSafetyStopChannel2();
+
+private:
+  /**
+   * @brief Blocking SDO read operation
+   *
+   * @exception std::runtime_error if operation fails
+   */
+  template <typename T>
+  T SyncSDORead(const std::uint16_t index, const std::uint8_t subindex);
```

Comment:
```suggestion
  T SyncSDORead(const CANopenObject & object);
```
this method can now use `CANopenObject` as a param

Reply:
check out comment below